src/lib/common/arg_macros.hpp

```diff
@@ -31,8 +31,8 @@
  *   This file includes macros for parsing variadic arguments.
  */
 
-#ifndef OT_ARG_MACROS_HPP_
-#define OT_ARG_MACROS_HPP_
+#ifndef OT_LIB_COMMON_ARG_MACROS_HPP_
+#define OT_LIB_COMMON_ARG_MACROS_HPP_
```

Comment:
How about instead of moving these common headers under `src/lib`, we move them under a new folder, `src/include`?

We want these include headers to be usable and shared across all the different OT components (`core`, `cli`, `ncp`, `posix`, `lib`). So, having them in their own `src/include` makes sense, and we can then update `COMMON_INCLUDES` in different CMake files to include `${PROJECT_SOURCE_DIR}/src/include` in addition to `${PROJECT_SOURCE_DIR}/src` and `${PROJECT_SOURCE_DIR}/src/core`.

The code in `lib` is mainly intended to be integrated into other projects (like the `spinel` library for use by a host), but we don't want to expose these common headers for use by any other software module (especially since they can cause macro name conflicts). So, placing them in their own `src/include` headers may be better than within `src/lib`.

Reply:
good idea!